server/models/database.go

```diff
@@ -79,41 +68,10 @@ func (d *DB) UpdatePassword(email string, password string) error {
 	return result.Error
 }
 
-// UpdateUserByID updates information of user
-func (d *DB) UpdateUserByID(id string, name string, password string, sshKey string, updatedAt time.Time, code int) (string, error) {
-	var res User
-	if name != "" {
-		result := d.db.Model(&res).Where("id = ?", id).Update("name", name)
-		if result.Error != nil {
-			return "", result.Error
-		}
-	}
-	if password != "" {
-		result := d.db.Model(&res).Where("id = ?", id).Update("hashed_password", password)
-		if result.Error != nil {
-			return "", result.Error
-		}
-	}
-	if sshKey != "" {
-		result := d.db.Model(&res).Where("id = ?", id).Update("ssh_key", sshKey)
-		if result.Error != nil {
-			return "", result.Error
-		}
-	}
-	if !updatedAt.IsZero() {
-		result := d.db.Model(&res).Where("id = ?", id).Update("updated_at", updatedAt)
-		if result.Error != nil {
-			return "", result.Error
-		}
-	}
-	if code != 0 {
-		result := d.db.Model(&res).Where("id = ?", id).Update("code", code)
-		if result.Error != nil {
-			return "", result.Error
-		}
-	}
-
-	return string(id), nil
+// UpdateUser updates information of user. empty and unchanged fields are not updated.
+func (d *DB) UpdateUserByID(id string, user User) (string, error) {
```

Comment:
it is optional, so if password is not sent from the enduser it will update the password to empty

Reply:
if password is not set it will be empty and won't be changed as mentioned in the comment